doc/reference/drivers/index.rst

```diff
@@ -33,13 +33,16 @@ are listed below.
 * **Serial communication**: This device driver is used by the kernel's
   system console subsystem.
 
-* **Random number generator**: This device driver provides a source of random
-  numbers.
+* **Entropy**: This device driver provides a source of entropy numbers
+  for the random number generator subsystem.
 
   .. important::
 
-    Certain implementations of this device driver do not generate sequences of
-    values that are truly random.
+    Use the random API functions for random values. Entropy functions
+    should not be directly used as a random number generator source as
+    certain hardware implementations are designed to be an entropy seed
+    source for random number generators and will have performance
+    problems trying to deliver a random stream of data.
```

Comment:
It's a deeper problem than this. Entropy sources are not suitable, in general, as a source of randomness. It isn't what they are providing. The outputs of these sources are not designed either for cryptographic purposes, or even pseudorandom purposes.

In other words, that some of them are slow is really the least of the problems with them.